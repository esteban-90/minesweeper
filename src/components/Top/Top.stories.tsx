import type { Story, Meta } from '@storybook/react'
import { Top, type TopProps } from './Top'

export default {
  title: 'Components/Top',
  component: Top,
} as Meta

const Template: Story<TopProps> = (args) => <Top {...args} />

export const TopExample = Template.bind({})

TopExample.args = {
  children: 'Minesweeper',
  feature: 'Flag',
  firstAction: 'ctrl',
  secondAction: 'click',
}
