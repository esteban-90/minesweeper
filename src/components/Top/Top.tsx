import type { FC } from 'react'
import styled from '@emotion/styled'
import { GameLegend, type GameLegendProps } from './GameLegend'
import { GameName, type GameNameProps } from './GameName'

const Header = styled.header`
  text-align: center;
  position: relative;
  display: inline-block;
`

export type TopProps = GameLegendProps & GameNameProps

export const Top: FC<TopProps> = ({ children: name, ...legendProps }): JSX.Element => {
  return (
    <Header>
      <GameName>{name}</GameName>
      <GameLegend {...legendProps} />
    </Header>
  )
}
