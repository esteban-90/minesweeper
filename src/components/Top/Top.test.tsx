import { render } from '@testing-library/react'
import { Top } from './Top'

describe('GameLegend test cases:', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Top feature='Flag' firstAction='ctrl' secondAction='click'>
        Minesweeper
      </Top>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
