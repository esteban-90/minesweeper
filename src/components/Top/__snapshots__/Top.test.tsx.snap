// Vitest Snapshot v1

exports[`GameLegend test cases: > should render correctly 1`] = `
<DocumentFragment>
  <header
    class="css-dn2347"
  >
    <h1
      class="css-18kvaxm"
    >
      Minesweeper
    </h1>
    <legend
      class="css-17qz6a2"
    >
      <strong>
        Flag:
      </strong>
      <code
        class="css-h9xasx"
      >
        <span
          class="css-udfepp"
        >
          ctrl
        </span>
         + 
        <span
          class="css-9c23k1"
        >
          click
        </span>
      </code>
    </legend>
  </header>
</DocumentFragment>
`;
